tests/config.rs

```diff
@@ -15,7 +15,7 @@ fn config_read_file_with_includes_ok() {
         got,
         Config {
             auth: Some(AuthMechanism::External),
-            listen: Some(Address::from_str("unix:path=/tmp/b").expect("should parse address")),
+            listen: Some(Address::from_str("unix:path=/tmp/a").expect("should parse address")),
```

Comment:
This last commit makes me a bit worried. Does this mean that we don't use the last listen address encountered? Although I wonder if that makes any diff in the typical (session and system) cases. :thinking:

Reply:
We do use the last listen address encountered

The issue is that these `<listen>`s are encountered via `<includedir>` which, according to dbus-daemon, is the equivalent of `<include>` for each of the files in that directory, but not in any specific order, and it turns out that `read_dir()` produces a different order of directory entries between my machine and the CI agent

```xml
<listen>1</listen>
<includedir><!-- expands into <listen>2</listen> --></includedir>
```

=>

```xml
<listen>1</listen>
<listen>2</listen>
```

The remaining test still asserts that an `<includedir></includedir>` following a `<listen>` still results in the expanded `<listen>` following the first `<listen>` and thus overriding it

It's just not useful or correct to assert that `<listen>`s inside sibling files that are referred to via `<includedir>` have any strict sequence one way or the other

Reply:
Ah it's about sibling files. As long as the spec doesn't say anything about this explicitly, I don't think we need to ensure any order.